### DISCUSSION TITLE
Panel-less traceback handler?

### DISCUSSION THREAD

**Discussion by pbsds** [Textualize/rich#2441] [General]
The border around the rich tracebacks is kind of annoying, as it gets in the way when i want to copy text, and it doesn't go well with terminal resizing. Is it possible to disable the panel/border of the traceback handler?

**Comment by ssbarnea** [comment]
I was looking for for something similar: removing the border on Markdown code blocks, ones that are also encapsulated inside a `Panel` and I was surprised to see that there is no empty/null `box`, so it is impossible to have a panel without borders... :p

**Comment by pradyunsg** [comment]
FWIW, it's relatively striaghtforward to create a `Box` that has no borders and use it with `Panel`:

```py
EMPTY = Box("    \n" * 8, ascii=True)
rich.print(Panel(Text("Hello!"), box=EMPTY))
```

It still has an additional padding of 1 character all around; since this is basically setting all the "box" characters to be... well... ` `.

Ideally, there'd be a way to pass `box` down from `Traceback` to `Panel`, and a `box.NOTHING` that served as a box that returned `''` values for all actual entries within `Box`. :)

**Comment by pradyunsg** [comment]
Cross-referencing https://github.com/Textualize/rich/discussions/2717#discussioncomment-4497561, where @willmcgugan has expressed willingness to accept a PR for the change as requested:

> I'd also accept a PR for a no-panel switch on Traceback. Probably not a "empty" box though, as that will add whitespace and I'll get folk complaining about that.

:)
